### PR TITLE
apply wdioLogLevel before starting workers

### PR DESCRIPTION
**Pull request by mgrybyk**
## Proposed changes

Want to change current behaviour of applying `logLevel`.

Current:
1. Initialise loggers with default `logLevel` (because config is not yet parsed)
2. Set `logLevels`
3. Set `logLevel` when workers start

As a result `logLevel` is ignored for some loggers and it is not possible to override `logLevel` in conf file using ENV variable.

New:
1. Initialise loggers with default `logLevel` (because config is not yet parsed)
2. Set `logLevel` if was not provided by user with ENV var `WDIO_LOG_LEVEL`
3. Set `logLevels`

In such case `logLevel` is applied before printing any messages to log

---

fixes some more issues mentioned in #3867 

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

Output example having only `logLevel` set to silent:

```
Test Suites:     1 passed, 1 total (100% completed)
Time:            🕐  8.02s
```

### CODE REVIEW

File: packages/wdio-logger/src/node.js [hunk 1]
```diff
@@ -126,20 +126,36 @@ export default function getLogger (name) {
...
+    Object.keys(loggers).forEach(logName => {
+        const logLevelName = getLogLevelName(logName)
 
-        if (logLevelsConfig[logLevelName] !== undefined) {
-            loggers[loggerName].setLevel(logLevelsConfig[logLevelName])
-        }
+        /**
+         * either apply log level from logLevels object or use global logLevel
+         */
+        const logLevel = logLevelsConfig[logLevelName] !== undefined ? logLevelsConfig[logLevelName] : process.env.WDIO_LOG_LEVEL
```
Comment (christian-bromann) [inline packages/wdio-logger/src/node.js]:
```suggestion
        const logLevel = logLevelsConfig[logLevelName] || process.env.WDIO_LOG_LEVEL
```

Comment (mgrybyk) [reply packages/wdio-logger/src/node.js]: @christian-bromann I'm afraid about case when `logLevelsConfig[logLevelName] === 0`.
In such case this ternary won't work 
`logLevelsConfig[logLevelName] || process.env.WDIO_LOG_LEVEL`

Comment (christian-bromann) [reply packages/wdio-logger/src/node.js]: yeah true, even tho it seems unlikely, how about doing

```
typeof logLevelsConfig[logLevelName] !== 'undefined'
```

I am kind of careful about comparing variables like this, e.g. `[undefined] == '' // true`

Comment (xlenz) [reply packages/wdio-logger/src/node.js]: I'll change to `typeof logLevelsConfig[logLevelName] !== 'undefined'` in the evening

**Pull request merged**